src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -683,12 +663,17 @@ private Task WriteHexInt32Async(int value)
                     // original information as the inner exception, for diagnostic purposes.
                     throw CancellationHelper.CreateOperationCanceledException(error, cancellationToken);
                 }
-                else if (error is InvalidOperationException || error is IOException)
+                else if (error is InvalidOperationException)
                 {
-                    // If it's an InvalidOperationException or an IOException, for consistency
-                    // with other handlers we wrap the exception in an HttpRequestException.
+                    // For consistency with other handlers we wrap the exception in an HttpRequestException.
                     throw new HttpRequestException(SR.net_http_client_execution_error, error);
                 }
+                else if (error is IOException ioe)
```

Comment:
I'm curious, why are InvalidOperationException and IOException different in this retry regard?

Reply:
This is just matching existing behavior. Previously we did the following in HttpConnectionPool:

```csharp
    catch (HttpRequestException e) when (!isNewConnection && e.InnerException is IOException && connection.CanRetry)
```